src/components/AppButton.vue

```diff
@@ -1,32 +1,34 @@
 <template>
-  <button :type="buttonType" :class="buttonIntent" @click="handleButtonClick">
+  <button :type="buttonType" :class="theme" @click="handleButtonClick">
     {{ title }}
   </button>
 </template>
 
 <script>
-import { CLOSE_APP_MODAL } from '@/store/mutation-types';
-import { mapMutations } from 'vuex';
-
 export default {
   name: 'AppButton',
   props: {
-    intent: { type: String, required: true },
+    isSubmit: {
+      type: Boolean,
+      default: false,
+    },
+    theme: {
+      type: String,
+      required: true,
+      validator: (val) => ['primary', 'secondary'].includes(val),
+    },
     title: { type: String, required: true },
   },
+  emits: ['closeModal'],
   computed: {
-    buttonIntent() {
-      return this.intent === 'cancel' ? 'cancel' : 'save';
-    },
     buttonType() {
-      return this.intent === 'create' ? 'submit' : 'button';
+      return this.isSubmit ? 'submit' : 'button';
     },
   },
   methods: {
-    ...mapMutations({ closeAppModal: CLOSE_APP_MODAL }),
     handleButtonClick() {
-      if (this.intent === 'cancel') {
-        this.closeAppModal();
+      if (!this.isSubmit) {
```

Comment:
It would be best if this component just emitted an event and a parent component that handles logic did the necessary action